install/minikube-installation-guide.md

```diff
@@ -0,0 +1,240 @@
+
+```sh
+$ kubectl get nodes
+NAME       STATUS   ROLES    AGE   VERSION
+minikube   Ready    master   78m   v1.13.4
+```
+
+Check you have virtualization enabled inside your Minikube:
+
+```sh
+# No color option, as minikube is running busybox
+$ minikube ssh "egrep 'vmx|svm' /proc/cpuinfo"
+
+# if you get a long line of output here, you have it enabled!
+flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon rep_good nopl xtopology cpuid pni pclmulqdq vmx ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch cpuid_fault invpcid_single pti retpoline rsb_ctxsw tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt xsaveopt xsavec xgetbv1 xsaves arat umip
+
+# A failed result means it is not enabled:
+ssh: Process exited with status 1
+```
+
+## Installing Kata Containers
+
+You can now install the Kata Containers runtime components. You will need a local copy of some Kata
+Containers components to help with this, and then use the host `kubectl` (that Minikube has already
+configured for you) to deploy them:
+
+```sh
+$ git clone https://github.com/kata-containers/packaging.git
+$ cd packaging/kata-deploy
+$ kubectl apply -f kata-rbac.yaml
+$ kubectl apply -f kata-deploy.yaml
+```
+
+This installs the Kata Containers components into `/opt/kata` inside the Minikube node. It can take
+a few minutes for the operation to complete. You can check the installation has worked by checking
+the status of the `kata-deploy` pod. This can be accomplished by running the following:
+
+```sh
+$ podname=$(kubectl -n kube-system get pods -o=name | fgrep kata-deploy | sed 's?pod/??')
+# If 'sleep infinity' is running, the pod has completed its work.
+$ kubectl -n kube-system exec ${podname} -- ps -ef | fgrep infinity
+root        49     1  0 10:55 ?        00:00:00 sleep infinity
+```
+
+> *NOTE:* This check only works for single node clusters, which is the default for Minikube.
+> For multi-node clusters, the check would need to be adapted to check `kata-deploy` had
+> completed on all nodes.
+
+## Enabling Kata Containers
+
+> **Note:** This step is only required for Minikube/Kubernetes versions <= 1.13. Since version
+> v1.14, the `RuntimeClass` is enabled by default. Performing this step on Kubernetes > v1.14 is
+> however benign.
+
+Now the Kata Containers components are installed in the Minikube node, you need to configure
+Kubernetes `RuntimeClass` to know when to use Kata Containers to run a pod.
+
+```sh
+$ curl https://raw.githubusercontent.com/kubernetes/node-api/master/manifests/runtimeclass_crd.yaml > runtimeclass_crd.yaml
+$ kubectl apply -f runtimeclass_crd.yaml
+```
+
+### Register the runtime
+
+Now register the `kata qemu` runtime with that class:
+
+```sh
+$ cd packaging/kata-deploy/k8s-1.14
```

Comment:
Will this doc go stale when a new version comes out?

Reply:
not really; we just needed pre-1.14, and post 1.14, since the feature became beta in 1.14 and updated the podSpec.

Seperate question though -- @grahamwhaley do we need to clone the packaging repository?  Can't we just kubectl apply -f <raw github link to packaging repos kata-deploy artifacts/kata-deploy.yaml?

Reply:
I'm sure we discussed this elsewhere before as well - the v1.14 is really '>=v1.14', so should work for all future installs.

Yep, you can kubectl from a URL - I'll see if I can change all the necessary occurrences and avoid any clone/curl/wgets